File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/responses/coupon/CouponResponse.java
package com.api.nature_harvest_backend.responses.coupon;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CouponResponse {
    private Long id;
    private String code;
    private String discountType;
    private String description;
    private String attribute;
    private String operator;
    private String value;
    private int discountAmount;
    private String startDate;
    private String endDate;
}

File: nature-harvest-frontend/src/app/responses/coupon/coupon.response.ts
export class CouponResponse {
  id: number;
  code: string;
  discountType: string;
  description: string;
  attribute: string;
  operator: string;
  value: string;
  discountAmount: number;
  startDate: string;
  endDate: string;

  constructor(
    id: number,
    code: string,
    discountType: string,
    description: string,
    attribute: string,
    operator: string,
    value: string,
    discountAmount: number,
    startDate: string,
    endDate: string
  ) {
    this.id = id;
    this.code = code;
    this.discountType = discountType;
    this.description = description;
    this.attribute = attribute;
    this.operator = operator;
    this.value = value;
    this.discountAmount = discountAmount;
    this.startDate = startDate;
    this.endDate = endDate;
  }
}
